rest_api/common/src/paging/v1/builder.rs

```diff
@@ -0,0 +1,138 @@
+// Copyright 2018-2022 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::marker::PhantomData;
+
+use super::{Paging, DEFAULT_LIMIT, DEFAULT_OFFSET};
+
+pub struct PagingBuilder<HasLink, HasQueryCount> {
```

Comment:
If we are going to use generics to enforce correctness, then the types should not be Yes/No but rather be containers with the values themselves, to avoid needing to unwrap during build().